tests/multigrid/mg_data_out_01.cc

```diff
@@ -0,0 +1,83 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2006 - 2016 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+// Test SingleLevelDataOut
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/tria.h>
+#include <deal.II/grid/tria_accessor.h>
+#include <deal.II/grid/tria_iterator.h>
+
+#include <deal.II/multigrid/mg_data_out.h>
+
+#include "../tests.h"
+
+using namespace dealii;
+
+template <int dim>
+void
+do_test()
+{
+  Triangulation<dim> triangulation(
+    Triangulation<dim>::limit_level_difference_at_vertices);
+  GridGenerator::hyper_cube(triangulation);
+  triangulation.refine_global(1);
+  triangulation.begin_active()->set_refine_flag();
+  triangulation.execute_coarsening_and_refinement();
+
+  deallog << "dim= " << dim << " cells=" << triangulation.n_cells()
+          << std::endl;
+
+  {
+    SingleLevelDataOut<dim> data_out(0);
+    data_out.attach_triangulation(triangulation);
+
+    deallog << "* level=0" << std::endl;
+
+    for (auto cell = data_out.first_cell();
+         cell.state() == IteratorState::valid;
+         cell = data_out.next_cell(cell))
+      {
+        deallog << " cell lvl=" << cell->level() << " active=" << cell->active()
+                << " id=" << cell->id().to_string() << std::endl;
+      }
+
+    for (unsigned int level = 0; level < triangulation.n_levels(); ++level)
+      {
+        deallog << "* level=" << level << std::endl;
+
+        data_out.reinit(level);
+
+        for (auto cell = data_out.first_cell(); cell != triangulation.end();
+             cell      = data_out.next_cell(cell))
+          {
+            deallog << " cell lvl=" << cell->level()
+                    << " active=" << cell->active()
+                    << " id=" << cell->id().to_string() << std::endl;
```

Comment:
Well, the `data_out` doesn't actually do anything. What's the purpose of the class and why shouldn't we test *that*?